### PR TITLE
made solution:

### PR DESCRIPTION
**Pull request by Sashk52**

### CODE REVIEW

File: src/main/java/core/basesyntax/Calculator.java [hunk 1]
```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+public class Calculator {
+    public double calculate(double firstDigit, double secondDigit, char operator) {
+        double result;
+        switch (operator) {
+            case '+':
+                result = firstDigit + secondDigit;
```
Comment (ShemsheyMaxim) [inline src/main/java/core/basesyntax/Calculator.java:8]: As for me you can do separate private method for operation and use this method in switch

File: src/main/java/core/basesyntax/Calculator.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+                break;
+            case '-':
+                result = firstDigit - secondDigit;
```
Comment (ShemsheyMaxim) [inline src/main/java/core/basesyntax/Calculator.java:11]: same

File: src/main/java/core/basesyntax/Calculator.java [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                break;
+            case '*':
+                result = firstDigit * secondDigit;
```
Comment (ShemsheyMaxim) [inline src/main/java/core/basesyntax/Calculator.java:14]: same

File: src/main/java/core/basesyntax/Calculator.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+                break;
+            case '/':
+                if (secondDigit == 0) {
+                    throw new ArithmeticException("Dividing by zero forbidden");
+                } else {
+                    result = firstDigit / secondDigit;
+                }
```
Comment (ShemsheyMaxim) [inline src/main/java/core/basesyntax/Calculator.java:21]: same

File: src/main/java/core/basesyntax/Calculator.java [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+                break;
+            case '^':
+                if (firstDigit < 0 && secondDigit < 1) {
+                    throw new ArithmeticException("Negative number can't "
+                            + "be raised to power less than 1 ");
+                } else {
+                    result = Math.pow(firstDigit, secondDigit);
```
Comment (ShemsheyMaxim) [inline src/main/java/core/basesyntax/Calculator.java:28]: same

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 6]
```diff
@@ -0,0 +1,330 @@
...
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class CalculatorTest {
+    private static Calculator calculator;
+
```
Comment (ShemsheyMaxim) [inline src/test/java/core/basesyntax/CalculatorTest.java:13]: You can use constant for magic nambers. I mean you can create constants such as POSITIVE_NUMBER, NEGATIVE_NUMBER etc. and use them in your test

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+        assertEquals(actual, expected, "Failed dividing two negative numbers");
+    }
+
+    @Test
+    void divideFirstZeroTestTest_Ok() {
+        double actual = calculator.calculate(0, 356, '/');
+        double expected = 0;
+        assertEquals(actual, expected, "Failed dividing first zero positive number");
+    }
+
```
Comment (ShemsheyMaxim) [inline src/test/java/core/basesyntax/CalculatorTest.java:218]: you can add test for check divide Number on Zero and throw exception

**Review by ShemsheyMaxim — APPROVED** [review]

File: src/main/java/core/basesyntax/Calculator.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
+package core.basesyntax;
+
+public class Calculator {
+    public double calculate(double firstDigit, double secondDigit, char operator) {
+        double result;
+        switch (operator) {
+            case '+':
+                result = firstDigit + secondDigit;
+                break;
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Calculator.java:9]: can we remove `result` variable and just use `return` in cases?

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 9]
```diff
@@ -0,0 +1,330 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class CalculatorTest {
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:11]: please add access modifiers to the class and all the methods in it

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 10]
```diff
@@ -0,0 +1,330 @@
...
+                 Double.POSITIVE_INFINITY, '+');
+        double expected = 0.0;
+        assertNotEquals(actual, expected);
+    }
+
+    @Test
+    void additingMinAndMaxVaueTest_Ok() {
+        double actual = calculator.calculate(Double.MIN_VALUE, Double.MIN_VALUE, '+');
+        double exprected = 1.0E-323;
+        assertEquals(actual, exprected, "Failed additing MinValue and MaxValue");
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:73]:
```suggestion
        assertEquals(actual, expected, "Failed adding MinValue and MaxValue");
```

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 11]
```diff
@@ -0,0 +1,330 @@
...
+        double expected = 50;
+        assertEquals(actual, expected, "Failed dividing two negative numbers");
+    }
+
+    @Test
+    void divideFirstZeroTestTest_Ok() {
+        double actual = calculator.calculate(0, 356, '/');
+        double expected = 0;
+        assertEquals(actual, expected, "Failed dividing first zero positive number");
+    }
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:217]: let's also add test where we divide by 0

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 12]
```diff
@@ -0,0 +1,330 @@
...
+        assertFalse(actualExcrection == notExpertedExctraction);
+        double actualMultiply = calculator.calculate(3, 45, '*');
+        double notExpertedMutliply = 48;
+        assertFalse(actualMultiply == notExpertedMutliply);
+        double actualDivide = calculator.calculate(30, 3, '/');
+        double notExpertedDivide = 15;
+        assertFalse(actualDivide == notExpertedDivide);
+        double actualRisePower = calculator.calculate(2, 3, '^');
+        double notExpertedRisePower = 6;
+        assertFalse(actualRisePower == notExpertedRisePower);
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:328]:
```suggestion
        double actualAddition = calculator.calculate(123, 456, '+');
        double notExpectedAddition = 123456;
        assertFalse(actualAddition == notExpectedAddition);
        double actualSubtraction = calculator.calculate(123456, 123, '-');
        double notExpectedSubtraction = 456;
        assertFalse(actualSubtraction  == notExpectedSubtraction );
        double actualMultiply = calculator.calculate(3, 45, '*');
        double notExpectedMutliply = 48;
        assertFalse(actualMultiply == notExpectedMutliply);
        double actualDivide = calculator.calculate(30, 3, '/');
        double notExpectedDivide = 15;
        assertFalse(actualDivide == notExpectedDivide);
        double actualRisePower = calculator.calculate(2, 3, '^');
        double notExpectedRisePower = 6;
        assertFalse(actualRisePower == notExpectedRisePower);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]
great job on the variety of tests! very nice

File: src/main/java/core/basesyntax/Calculator.java (hunk 1 above)
Comment (KseniiaMakarova) [reply src/main/java/core/basesyntax/Calculator.java]: @ShemsheyMaxim this is optional, but in this case we can go without this

File: src/main/java/core/basesyntax/Calculator.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
...
+            case '-':
+                return firstDigit - secondDigit;
+            case '*':
+                return firstDigit * secondDigit;
+            case '/':
+                if (secondDigit == 0) {
+                    throw new ArithmeticException("Dividing by zero forbidden");
+                } else {
+                    return firstDigit / secondDigit;
+                }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Calculator.java:17]: do we need else after `throw`?
```suggestion
                return firstDigit / secondDigit;
```

File: src/main/java/core/basesyntax/Calculator.java [hunk 14]
```diff
@@ -0,0 +1,29 @@
...
+            case '^':
+                if (firstDigit < 0 && secondDigit < 1) {
+                    throw new ArithmeticException("Negative number can't "
+                            + "be raised to power less than 1 ");
+                } else {
+                    return Math.pow(firstDigit, secondDigit);
+                }
```
Comment (KseniiaMakarova) [inline src/main/java/core/basesyntax/Calculator.java:24]: same here

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 15]
```diff
@@ -0,0 +1,337 @@
...
+        assertFalse(actualSubtraction == notExpectedSubtraction);
+        double actualMultiply = calculator.calculate(3, 45, '*');
+        double notExpectedMutliply = 48;
+        assertFalse(actualMultiply == notExpectedMutliply);
+        double actualDivide = calculator.calculate(30, 3, '/');
+        double notExpectedDivide = 15;
+        assertFalse(actualDivide == notExpectedDivide);
+        double actualRisePower = calculator.calculate(2, 3, '^');
+        double notExpectedRisePowe = 6;
+        assertFalse(actualRisePower == notExpectedRisePowe);
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:335]:
```suggestion
        double notExpectedRisePower = 6;
        assertFalse(actualRisePower == notExpectedRisePower);
```

File: src/test/java/core/basesyntax/CalculatorTest.java [hunk 16]
```diff
@@ -0,0 +1,337 @@
...
+        assertThrows(ArithmeticException.class, () -> {
+            calculator.calculate(-25, 0.5, '^');
+        });
+    }
+
+    @Test
+    public void falseCasesTests_NotOk() {
+        double actualAddition = calculator.calculate(123, 456, '+');
+        double notExpertedAddition = 123456;
+        assertFalse(actualAddition == notExpertedAddition);
```
Comment (KseniiaMakarova) [inline src/test/java/core/basesyntax/CalculatorTest.java:323]:
```suggestion
        double notExpectedAddition = 123456;
        assertFalse(actualAddition == notExpectedAddition);
```

**Review by KseniiaMakarova — CHANGES_REQUESTED** [review]

**Review by KseniiaMakarova — APPROVED** [review]
